executor/aggfuncs/func_count.go

```diff
@@ -0,0 +1,99 @@
+package aggfuncs
+
+import (
+	"github.com/juju/errors"
+
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/util/chunk"
+)
+
+type baseCount struct {
+	baseAggFunc
+}
+
+type partialResult4Count = int64
+
+func (e *baseCount) AllocPartialResult() PartialResult {
+	return PartialResult(new(partialResult4Count))
+}
+
+func (e *baseCount) ResetPartialResult(pr PartialResult) {
+	p := (*partialResult4Count)(pr)
+	*p = 0
+}
+
+func (e *baseCount) AppendFinalResult2Chunk(sctx sessionctx.Context, pr PartialResult, chk *chunk.Chunk) error {
+	p := (*partialResult4Count)(pr)
+	chk.AppendInt64(e.ordinal, *p)
+	return nil
+}
+
+type countOriginal struct {
+	baseCount
+}
+
+func (e *countOriginal) UpdatePartialResult(sctx sessionctx.Context, rowsInGroup []chunk.Row, pr PartialResult) error {
+	p := (*partialResult4Count)(pr)
+
+	for _, row := range rowsInGroup {
+		_, isNull, err := e.args[0].EvalInt(sctx, row)
```

Comment:
We should implement countOriginal4Int/ countOriginal4Real/ countOriginal4String/ countOriginal4Decimal/ countOriginal4Datetime/ countOriginal4Duration/ countOriginal4Json. 
Since it will cause an error if we call EvalInt on other types.

Reply:
I will fix it.